example/simple_testing/c/Wrapper.h

```diff
@@ -0,0 +1,170 @@
+#ifndef Wrapper_H
+#define Wrapper_H
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "diplomat_runtime.h"
+
+#include "TestingStruct.d.h"
+
+#include "Wrapper.d.h"
+
+
+
+
+
+
+typedef struct DiplomatCallback_Wrapper_test_multi_arg_callback_f {
+    const void* data;
+    int32_t (*run_callback)(int32_t(*cb_pointer)(int32_t), int32_t );
```

Comment:
question: I don't quite understand why there is a nested function here

(corollary: don't understand why we need the convenience functions below)

Reply:
simplified/removed! good point, that was my bad for not realizing how these functions are typically provided